### ISSUE TITLE
Documents directory is removed entirely on app launch

### ISSUE THREAD

**Issue by simonmitchell** [joltup/rn-fetch-blob#204]
Library version: 0.10.13
RN version: 0.57.0

Sample code can be seen here:
```
Promise.all(
	tableOfContents.map((content, index) => {
		let path = `${dirs.DocumentDir}/${childId}/page_${index}.jpg`;
		let pageURL = content.url;
		return RNFetchBlob.config({
			path: path
		}).fetch("GET", pageURL).then(res => {
			// Set session to child ID so we can clear these all out
			try {
				res.session(childId);
			} catch (error) {
				console.info("Error setting download session", error);
			}

			return res.data;
		});
	})
)
```

When the app is re-run from Xcode, the documents directory (not the sub-directory we create) is deleted from the device (be it a simulator or a physical device). The docs state that cache is not cleared unless it’s done so manually, is there something wrong I’m doing here?

**Comment by simonmitchell** [comment]
Think I've got to the bottom of this. It seems that this isn't actually occurring on app launch, rather on app update (Which in the simulator is every time you run from Xcode). The problem being that the documents directory's contents remain intact on update, but it's root location changes. This means that the absolute paths that I'm utilising are incorrect on update, so I should just be saving the path extension to state and re-constructing using `dirs.DocumentDir` each time I need to access a file!

**Comment by bneigher** [comment]
@simonmitchell do you think this is a bad practice?  Maybe there is other caching data that you'll be preserving over time as you update the app multiple times and yet keep the data.

I would imagine the OS cleans up old app version caches over time on purpose

**Comment by sankar9659** [comment]
Any update on this issue? still getting this error.

**Comment by bneigher** [comment]
@sankar9659 et all.  While frustrating - I've settled with the fact that this is kind of outside the scope of this library.  Let me explain why and what I did to get around this "limitation"

Each time the app updates with a new build, there is some internal changes to the device storage / app reference in the filesystem which changes the path the these directories (camera, document, ect.)  This is probably a security feature that the OS has forced for apps.  What essentially happens is that if you're storing PATHS to files, you gonna have a bad time.  This should be a disclaimer for react native developers to know before doing something like storing local paths.

YOU SHOULD either only store relative paths and calculate the absolute prefix all around your app, or if that ain't going to cut it - devise a migration strategy to update paths to files and update the paths each time there is a new build / version of your app.  I did the latter because I would need to rewrite my app if I wanted to rely on the former.

tldr;  You can't rely on absolute paths to be stored for local files.  Use the library to generate the prefix (or the entire path) for the file because files will be moving around on the device when the app updates.

**Comment by sankar9659** [comment]
@bneigher Then there is no way to get the removed files, after update an app, right?
or Have you found any solution for this?
If you found the solution please let me know or send the code.

**Comment by bneigher** [comment]
@sankar9659 the files are not deleted, they are just moved.  You need to modify existing paths manually.

Roughly:

Before app update:
`/this/is/the/oldpath/uuid_old/documents/myfile.jpg`

After app update:
`/this/is/the/newpath/uuid_new/documents/myfile.jpg`

Recommendation for developers:

Always store paths as their relative path:
`myfile.jpg`

And use this library to generate the prefix:
`${RNFetchBlob.fs.dirs.DocumentDir}/myfile.jpg`

**Comment by sankar9659** [comment]
@bneigher  Here is my code.
`const downloadImage = () => {
        const { config, fs } = RNFetchBlob;
        RNFetchBlob
            .config({
                // response data will be saved to this path if it has access right.
                path: `${RNFetchBlob.fs.dirs.DocumentDir}/myfile.jpg`,
            })
            .fetch('GET', 'http://www.example.com/file/example.zip', {
                //some headers ..
            })
            .then((res) => {
                AsyncStorage.setItem('localImage', JSON.stringify(res.path()))
                 setLocalImage(res.path())
                alert('Image Downloaded Successfully.');
                console.log('The file saved to ', res.path())
            })
    }; `

I'm trying
 to show the image using the path but getting this error.
Could not find image file:///Users/mitrah_mac_2/Library/Developer/CoreSimulator/Devices/F8FBE22B-D859-462B-83D3-EF8551F36133/data/Containers/Data/Application/707AB961-4E63-4C63-9872-9A1457AFF1F3/Documents/myfile.jpg

Anything wrong in this code?

**Comment by sankar9659** [comment]
@bneigher Issue fixed thanks. ;}

import React, { useState, useEffect } from 'react';

import * as RNFS from 'react-native-fs';

// Import Required Components
import {
    StyleSheet,
    Text,
    View,
    TouchableOpacity,
    PermissionsAndroid,
    Image,
    Platform,
} from 'react-native';

import AsyncStorage from '@react-native-async-storage/async-storage';

const Settings = () => {
    const [LocalImage, setLocalImage] = useState('')


    const image_URL =
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQZ6QSUYDRBRChIXGs0jqH55cZytOPPjjh4Bg&usqp=CAU'

    const renderImagePath = async () => {
        let initialImage = await AsyncStorage.getItem('localImage')
        if (initialImage == null) {
            const extension = (Platform.OS === 'android') ? 'file://' : ''
            const path = `${extension}${RNFS.DocumentDirectoryPath}/localImg.png`; //U can use any format png, jpeg, jpg

            RNFS.exists(path).then(exists => {
                RNFS.downloadFile({ fromUrl: image_URL, toFile: path }).promise.then(res => {
                    setLocalImage(`${RNFS.DocumentDirectoryPath}/localImg.png`)
                    AsyncStorage.setItem('localImage', 'localImg.png')
                })
            }).catch(error => {
                console.warn(error);
            });
        }
        else {
            console.log('file exists', initialImage)
            setLocalImage(`${RNFS.DocumentDirectoryPath}/${initialImage}`)
        }
    }

    console.log('LocalImage', LocalImage)

    return (
        <View style={styles.container}>
            <View style={{ alignItems: 'center' }}>
                <Text style={{ fontSize: 30, textAlign: 'center' }}>
                    React Native Image Download Example
        </Text>
                <Text
                    style={{
                        fontSize: 25,
                        marginTop: 20,
                        marginBottom: 30,
                        textAlign: 'center',
                    }}>
                    www.aboutreact.com
        </Text>
            </View>
            {LocalImage ?
                <Image
                    source={{
                        uri: `file://${LocalImage}`
                    }}
                    style={{
                        width: '100%',
                        height: 100,
                        resizeMode: 'contain',
                        margin: 5
                    }}
                />
                : null}
            <TouchableOpacity
                style={styles.button}
                onPress={renderImagePath}>
                <Text style={styles.text}>
                    Download Image
        </Text>
            </TouchableOpacity>
        </View>
    );
};

export default Settings;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    },
    button: {
        width: '80%',
        padding: 10,
        backgroundColor: 'orange',
        margin: 10,
    },
    text: {
        color: '#fff',
        fontSize: 20,
        textAlign: 'center',
        padding: 5,
    },
});
